lib/private/app/appmanager.php

```
@@ -279,4 +281,40 @@ public function getIncompatibleApps($version) {
 		return $incompatibleApps;
 	}
 
+	/**
+	 * @inheritdoc
+	 */
+	public function isShipped($appId) {
+		$this->loadShippedJson();
+		return in_array($appId, $this->shippedApps);
+	}
+
+	private function isAlwaysEnabled($appId) {
+		$alwaysEnabled = $this->getAlwaysEnabledApps();
+		return in_array($appId, $alwaysEnabled);
+	}
+
+	private function loadShippedJson() {
+		if (is_null($this->shippedApps)) {
+			$shippedJson = \OC::$SERVERROOT . '/core/shipped.json';
+			if (file_exists($shippedJson)) {
+				$content = json_decode(file_get_contents($shippedJson), true);
+				$this->shippedApps = $content['shippedApps'];
+				$this->alwaysEnabled = $content['alwaysEnabled'];
+			} else {
```

IMHO fail hard. No need to have another code path from my PoV. Also the code check to be introduced with 9.0 will require this file to be available for a pass.